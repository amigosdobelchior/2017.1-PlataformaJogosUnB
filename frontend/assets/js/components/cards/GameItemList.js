import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {Image, Item, Button, Header,Icon} from "semantic-ui-react";
import {getKernel} from "../../resources/getKernel"
import {kernelValidation} from "../../resources/kernelValidation"

const cardImageStyle = {
    "background": "#000000",
    "position": "relative",
    "minHeight": "200px",
    "minWidth": "200px",
};
const imageStyle = {
    "position": "absolute",
    "top": 0,
    "bottom": 0,
    "right": 0,
    "left": 0,
    "margin": "auto",
    "height": "100%",
};

export default class GameItemList extends React.Component {

    getGameLink(element){
        return (
            <Link to={`/games/${this.props.game.pk}/${this.props.game.name}`}
                params={{"id": this.props.game.pk}}>
                {element}
            </Link>
        )
    }
    
    mountIcons(kernels){
        return kernels.map((kernel) => {
             return (<Icon inverted size={kernelValidation(kernel)} className={kernel} />)

        })
    }

    render () {
        return (
            <Item.Group divided unstackable>
                <Item>
                  <Item.Image size='tiny' style={cardImageStyle}>
                    {this.getGameLink(<Image src={this.props.game.card_image} style={imageStyle,cardImageStyle} />)}
                  </Item.Image>
                  <Item.Content verticalAlign='middle'>
                    <Item.Header>
                        {this.getGameLink(<Header inverted>{this.props.game.name}</Header>)}
                    </Item.Header>
                    
                    
                    <Item.Extra>
                        {this.getGameLink(<Button basic color='green' floated='right'>Conhe√ßa mais...</Button>)}
                        
                    </Item.Extra>
                  
                    <Item.Extra>
                           <div style={{right:0, position:"absolute"}}>
                           {this.mountIcons(getKernel(this.props.game.packages))}
</div>
                    </Item.Extra>
                  </Item.Content>
                </Item>
            </Item.Group>
        );

    }
}

GameItemList.propTypes = {
    game: PropTypes.object.isRequired,
    reducePlatforms: PropTypes.func.isRequired,
}
