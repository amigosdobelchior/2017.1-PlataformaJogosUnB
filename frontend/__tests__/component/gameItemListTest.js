import React from 'react';
import GameItemList from '../../assets/js/components/cards/GameItemList';
import {Card, Image, Item, Label, Button,Icon, Header} from "semantic-ui-react";
import {Link} from "react-router-dom";
import renderer from 'react-test-renderer';

test('Test render GameItemList', () => {
  const component = renderer.create(
      <Item.Group divided unstackable>
          <Item>
            <Item.Image size='tiny' style="style">
             <Image src="./image.jpg" style="style" />)
            </Item.Image>
            <Item.Content verticalAlign='middle'>
              <Item.Header>
                  "name"
              </Item.Header>

              <Item.Meta>
                  <Image.Group>

                         <Image key={1} src={1} width='30' height='30' /> )

                  </Image.Group>
              </Item.Meta>

              <Item.Extra>
                 <Button basic color='green' floated='right'>Baixe agora!</Button>)

                 <Label color='green'> "acao" </Label>

              </Item.Extra>

            </Item.Content>
          </Item>
      </Item.Group>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
