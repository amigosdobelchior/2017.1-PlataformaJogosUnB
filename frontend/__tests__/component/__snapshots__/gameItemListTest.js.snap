// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render GameItemList 1`] = `
<div
  className="ui divided unstackable items"
>
  <div
    className="item"
  >
    <div
      className="ui tiny image"
      style="style"
    >
      <img
        alt={undefined}
        className="ui image"
        height={undefined}
        src="./image.jpg"
        style="style"
        width={undefined}
      />
      )
    </div>
    <div
      className="middle aligned content"
    >
      <div
        className="header"
      >
        name
      </div>
      <div
        className="meta"
      >
        <div
          className="ui images"
        >
          <img
            alt={undefined}
            className="ui image"
            height="30"
            src={1}
            width="30"
          />
           )
        </div>
      </div>
      <div
        className="extra"
      >
        <button
          className="ui green basic right floated button"
          onClick={[Function]}
          tabIndex={undefined}
        >
          Baixe agora!
        </button>
        )
        <div
          className="ui green label"
          onClick={[Function]}
        >
          acao
        </div>
      </div>
    </div>
  </div>
</div>
`;
